pyanaconda/kickstart.py

```diff
@@ -653,14 +653,13 @@ def parseKickstart(handler, f, strict_mode=False, pass_to_boss=False):
     ksparser = AnacondaKSParser(handler)
     kswarnings = []
     ksmodule = "pykickstart"
-    kscategories = (UserWarning, SyntaxWarning, DeprecationWarning)
     showwarning = warnings.showwarning
 
     def ksshowwarning(message, category, filename, lineno, file=None, line=None):
         # Print the warning with default function.
         showwarning(message, category, filename, lineno, file, line)
         # Collect pykickstart warnings.
-        if ksmodule in filename and issubclass(category, kscategories):
+        if ksmodule in filename and issubclass(category, KickstartParseWarning):
             kswarnings.append(message)
```

Comment:
I would investigate whether we can use `warnings.catch_warnings(record=True)` to collect warnings.